tasks/kmt.py

```diff
@@ -358,27 +474,48 @@ def test(ctx, vms, stack=None, packages="", run=None, retry=2, rebuild_deps=Fals
         json.dump(run_config, tmp)
         tmp.flush()
 
+        args = [
+            f"-packages-run-config {tmp.name}",
+            f"-retry {retry}",
+            "-verbose" if test_logs else "",
+            f"-run-count {run_count}",
+            "-test-root /opt/system-probe-tests",
+        ]
         for d in domains:
             d.copy(ctx, f"{tmp.name}", "/tmp")
-            d.run_cmd(ctx, f"bash /micro-vm-init.sh {retry} {tmp.name}", verbose=verbose)
+            d.run_cmd(ctx, f"bash /micro-vm-init.sh {' '.join(args)}", verbose=verbose)
 
 
-@task
-def build(ctx, vms, stack=None, ssh_key=None, rebuild_deps=False, verbose=True):
+@task(
+    help={
+        "vms": "Comma seperated list of vms to target when running tests",
+        "stack": "Stack in which the VMs exist. If not provided stack is autogenerated based on branch name",
+        "ssh-key": "SSH key to use for connecting to a remote EC2 instance hosting the target VM",
+        "full-rebuild": "Do a full rebuild of all test dependencies to share with VMs, before running tests. Useful when changes are not being picked up correctly",
+        "verbose": "Enable full output of all commands executed",
+    }
+)
+def build(ctx, vms, stack=None, ssh_key=None, full_rebuild=False, verbose=True):
     stack = check_and_get_stack(stack)
     if not stacks.stack_exists(stack):
         raise Exit(f"Stack {stack} does not exist. Please create with 'inv kmt.stack-create --stack=<name>'")
 
     if not os.path.exists(f"kmt-deps/{stack}"):
         ctx.run(f"mkdir -p kmt-deps/{stack}")
 
+    arch = platform.machine()
     infra = build_infrastructure(stack, ssh_key)
     domains = filter_target_domains(vms, infra)
```

Comment:
Same issue here as before, we might be copying binaries of the wrong architecture.